controllers/apps/transformer_component_restore.go

```diff
@@ -61,13 +60,8 @@ func (t *componentRestoreTransformer) Transform(ctx graph.TransformContext, dag
 	cluster := transCtx.Cluster
 	restoreMGR := plan.NewRestoreManager(reqCtx.Ctx, t.Client, cluster, rscheme, nil, synthesizedComp.Replicas, 0)
 
-	actionCtx, err := component.NewActionContext(cluster, transCtx.Component, transCtx.RunningWorkload,
-		synthesizedComp.LifecycleActions, synthesizedComp.ScriptTemplates, component.PostProvisionAction)
-	if err != nil {
-		return err
-	}
-	needDoPostProvision, _ := component.NeedDoPostProvision(transCtx.Context, transCtx.Client, actionCtx)
-	if err := restoreMGR.DoRestore(synthesizedComp, transCtx.Component, needDoPostProvision); err != nil {
+	postProvisionDone := checkPostProvisionDone(transCtx)
```

Comment:
why not do the check outside of restoreMGR.DoRestore?

if ! postProvisionDone {
    return nil
}

Reply:
@wangyelei is that ok?